### ISSUE TITLE
Big acc difference when fine-tune using vgg16 and resnet-50

### ISSUE THREAD

**Issue by Paul0M** [apache/incubator-mxnet#11040]
Test this https://mxnet.incubator.apache.org/faq/finetune.html

when using **resnet-50 and flatten0**, easy to get acc > **80%**, (sgd, lr=0.01)

when using **Inception-BN and flatten**, easy to get acc > **７０%**, (sgd, lr=0.1)

But when using **vgg16 and flatten_0 or fc8**, only get acc over **10%**, (tried sgd,adm with different lr)

Does any one have any idea about this big difference?

**Comment by wsokolow** [comment]
Which engine did you use for execution? GPU or CPU (with MKLDNN)? I'm observing issues with ~10% accuracy while running AlexNet with MKLDNN engine.

**Comment by sandeep-krishnamurthy** [comment]
Hello, I would also encourage you to ask your question at https://discuss.mxnet.io/ for more people being able to answer, search and refer in future

**Comment by Paul0M** [comment]
@wsokolow GPU

**Comment by wsokolow** [comment]
I've also reproduced my problem with GPU.

**Comment by ThomasDelteil** [comment]
This is due to the fact that flatten_0 is not the penultimate layer in VGG16. See answer here: https://discuss.mxnet.io/t/issue-big-acc-difference-when-fine-tune-using-vgg16-and-resnet-50/1124

**Comment by ThomasDelteil** [comment]
You can check your network architecture like that:
```
mx.viz.print_summary(sym)
```
```
________________________________________________________________________________________________________________________
Layer (type)                                        Output Shape            Param #     Previous Layer                  
========================================================================================================================
data(null)                                                                  0                                           
________________________________________________________________________________________________________________________
conv1_1(Convolution)                                                        64          data                            
________________________________________________________________________________________________________________________
relu1_1(Activation)                                                         0           conv1_1                         
________________________________________________________________________________________________________________________
....
________________________________________________________________________________________________________________________
fc7(FullyConnected)                                                         4096        drop6                           
________________________________________________________________________________________________________________________
relu7(Activation)                                                           0           fc7                             
________________________________________________________________________________________________________________________
drop7(Dropout)                                                              0           relu7                           
________________________________________________________________________________________________________________________
fc8(FullyConnected)                                                         1000        drop7                           
________________________________________________________________________________________________________________________
prob(SoftmaxOutput)                                                         0           fc8                             
========================================================================================================================
Total params: 13416
________________________________________________________________________________________________________________________
```

You want to use `relu7` for your fine-tuning layer.
```
(new_sym, new_args) = get_fine_tune_model(sym, arg_params, num_classes, layer_name='relu7')
```

Result:
```
2018-05-28 18:57:46,908 Epoch[7] Batch [230]	Speed: 441.92 samples/sec	accuracy=0.884375
2018-05-28 18:57:48,405 Epoch[7] Batch [240]	Speed: 427.60 samples/sec	accuracy=0.934375
2018-05-28 18:57:48,406 Epoch[7] Train-accuracy=0.934375
2018-05-28 18:57:48,406 Epoch[7] Time cost=35.405
2018-05-28 18:58:00,677 Epoch[7] Validation-accuracy=0.725211
```

**Comment by ThomasDelteil** [comment]
@indhub can you close this issue? It has been resolved on the forum. Thanks!
